gh-87506: Document that `json.load*()` can raise `UnicodeDecodeError` (#127355)
python/cpython: Doc/library/json.rst

   The other arguments have the same meaning as in :func:`load`.

   If the data is not a valid JSON document, a :exc:`JSONDecodeError` will be

>    raised. If the binary file does not contain UTF-8, UTF-16 or UTF-32 encoded

Reviewer: There is no file here. In any case the wording differs from the suggested by @rhettinger, so I leave this on him.

Reviewer 2: This seems fine to me.

   :raises JSONDecodeError:
      When the data being deserialized is not a valid JSON document.

   : raises UnicodeDecodeError:

>       If the binary file does not contain UTF-8, UTF-16 or UTF-32 encoded data.

Reviewer 3: ```suggestion
      When the data being deserialized does not contain
      UTF-8, UTF-16 or UTF-32 encoded data.
```

   The other arguments have the same meaning as in :func:`load`.

   If the data is not a valid JSON document, a :exc:`JSONDecodeError` will be
   raised. If the binary file does not contain UTF-8, UTF-16 or UTF-32 encoded

>    data, a :exc:`UnicodeDecodeError` will be raised.

Reviewer 3: Let's just remove this; there is no need to duplicate what's in the parameter list nearby.

Reviewer 3: Sorry, I did not clearly communicate what I meant. I put up an amendment PR: #128609.
